gpii/node_modules/testing/src/Integration.js

```diff
@@ -371,3 +371,60 @@ fluid.defaults("gpii.test.integration.mockSettingsHandlerRegistry.android", {
         }
     }
 });
+
+fluid.defaults("gpii.test.integration.deviceReporterAware", {
+    gradeNames: ["fluid.eventedComponent", "autoInit"],
+    members: {
+        deviceReporters: {}
+    },
+    listeners: {
+        onCreate: "gpii.test.integration.deviceReporterAware.populate"
+    }
+});
+
+gpii.test.integration.deviceReporterAware.mockDeviceReporters = {};
+
+gpii.test.integration.deviceReporterAware.mockDeviceReporters.findNameInList = function (expectInstalled, name) {
+    return expectInstalled.some(function (el) { return el === name; });
+};
+
+gpii.test.integration.deviceReporterAware.populate = function (that) {
+    fluid.each(that.options.deviceReporters, function (options, key) {
+        if (fluid.getGlobalValue(key)) {
```

Comment:
The nameResolver should give a completely fresh name for the deviceReporter under the mock configuration - for example, the way our settingsHandlers do. There should never be a question of overwriting the global name of the real deviceReporter, etc.